can/interfaces/socketcan/socketcan.py

```diff
@@ -286,7 +287,18 @@ def send_bcm(bcm_socket: socket.socket, data: bytes) -> int:
             raise e
 
 
-def _add_flags_to_can_id(message: Message) -> int:
+GLOBAL_TASK_ID = 0
+TASK_LOCK = threading.Lock()
+
+
+def _compose_next_task_id() -> int:
+    with TASK_LOCK:
+        global GLOBAL_TASK_ID
+        GLOBAL_TASK_ID += 1
+        return GLOBAL_TASK_ID
```

Comment:
We will want to also ensure that we don't overflow the `canid_t` type, on the off-chance that someone is running this for extended periods of time and repeatedly starting/stopping periodic TX tasks.

```c
typedef __u32 canid_t;
```

Ensuring that we wrap at `2^11 - 1` is probably sufficient for now.

Reply:
Thanks, good catch. It looks to be an unsigned integer though, thus wrapping is required at `2^32 - 1`, isn't it?

Considering this accidental case, some of the previous tasks IDs can be used already, wouldn't it better still implement usage of `CAN_BCM_TX_READ` to check if they are busy or not?

Reply:
When I commented, I wasn't sure if the kernel code was doing anything with anything beyond the 11-bits (ie. checking other bits for various flags in the 32-bits) so I just said 11-bits to be safe to avoid any potential unintended side effects.

Looking more closely through the kernel code now, it seems like for TX Tasks, this value is only copied (for things other than the TX Task ID) when `TX_CP_CAN_ID` is set as part of `flags` on the BCM message. The only places it's used for determining behaviour is when dealing with RX tasks, but they're stored in separate lists so they won't affect each other (and `python-can` currently doesn't support this).

But yes, you're right, it should be safe to increase this to 32-bits.

Reply:
> Considering this accidental case, some of the previous tasks IDs can be used already, wouldn't it better still implement usage of CAN_BCM_TX_READ to check if they are busy or not?

Yes, let's do that. If we detect that the TX ID has been used, let's just raise an Exception, and make it explicit for the user to decide whether to retry calling `bus.send_periodic`.

Reply:
Done